### ISSUE TITLE
May ?start and ?end reference passages in different depths of a document?

### ISSUE THREAD

**Issue by kbrueckmann** [distributed-text-services/specifications#265]
I'm working with the latest version of the documentation right now (https://distributed-text-services.github.io/specifications/versions/unstable/#navigation-endpoint) and a bit confused about a phrase in the table "Usage of down, ref, start and end":

> A member array of CitableUnits including the citation tree between the start and end CitableUnits inclusive, down to a depth _relative to the deeper of the CitableUnits identified by start and end_.

If I understand this correctly, the two passages identified by start and end may be on different "levels" or "depths" in the document; e.g. start=C1 and end=C3.E1 (instead of just C1 and C3 as in example 6 of the specs). In our implementation (still in the works) we had planned on not allowing that as it might lead to dts:wrapper-contents that are not well-formed for the document endpoint (e.g. we return C1 and C2 with all their paragraph children, then start C3 but stop after a few paragraphs and thus have a missing closing tag). 

Looking at the other examples in the specs, I couldn't find one that includes start/end-values of different levels, so I hope it's just a small irregularity in that table I cited. If so, it might be useful to also explicitly state the restriction somewhere. If not, however, I'd love some hints about how others have implemented this :).

**Comment by PonteIneptique** [comment]
Hi @kbrueckmann :)

Yes, normally, range can have various depth, we might have removed the example from our documentation when we simplified the docs. I would prefer to have the confirmation of @hcayless or @monotasker in case I am wrong.

As to how to do this, the only implementation I have developed traverse the tree between start and end:
- while start and end share common parents, I copy the parent nodes.
- when they differ, I check if there are siblings between them. Typically, if i have div[n=2] as a start and div[n=5] as an end, I will continue treating div[n=2] "manually", continue treating div[n=5] manually, but in between both these nodes, I will copy the content plain and simple.
- if they differ in depth, it does not change much for me: I treat the XPath of start and the XPath of end separately.

I have been working on a reference implementation, and this mechanism is the only one implemented. 

https://github.com/distributed-text-services/MyDapytains

Note that my current implementation does not use dts:wrapper for now.
